client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/checklist-item.tsx

```diff
@@ -22,34 +22,45 @@ const ChecklistItem = ( { task }: { task: Task } ) => {
 				'not-completed': ! isCompleted && ! keepActive, // a task that hasn't been completed yet
 			} ) }
 		>
-			<Button
-				className="launchpad__checklist-item"
-				disabled={ taskDisabled }
-				data-task={ id }
-				onClick={ actionDispatch }
-			>
-				{ isCompleted && (
-					// show checkmark for completed tasks regardless if they are disabled or kept active
-					<div className="launchpad__checklist-item-checkmark-container">
+			{ isPrimaryAction ? (
```

Comment:
I like this idea a lot it's very neat

Reply:
Im on the fence with this idea as I think it has some pros/cons and doesn't necessarily buy us a lot? I don't anticipate there being many other forms of checklist item components other than the standard list items and the main CTA at the end.

Reply:
Im approving from my end for the current behavior and code, but feel free to update for the suggestions above if you think these make more sense @kirahsapong

Reply:
I agree with @Addison-Stavlo. it is a neat pattern for sure and works well if we require additional button variants. In the spirit of not pre-optimizing _too_ much, will be great to have as a follow-up in the event we add additional variants - this will be a stellar approach.